`skaffold init` supports different exit codes (#4408)
GoogleContainerTools/skaffold: docs/content/en/docs/pipeline-stages/init.md

Skaffold auto-generates `build` and `deploy` config for supported builders and deployers.

## Build Config Initialization

`skaffold init` currently supports build detection for those builders:

1. [Docker]({{<relref "/docs/pipeline-stages/builders/docker">}})
2. [Jib]({{<relref "/docs/pipeline-stages/builders/jib">}}) (with `--XXenableJibInit` flag)
2. [Buildpacks]({{<relref "/docs/pipeline-stages/builders/buildpacks">}}) (with `--XXenableBuildpacksInit` flag)

> `skaffold init` walks your project directory and looks for any `Dockerfile`s 
> , `build.gradle/pom.xml`, `package.json`, `requirements.txt` or `go.mod`. Please note,
> it'll skip files that are larger than 500MB.

Reviewer: ```suggestion
`skaffold init` walks your project directory and looks for any build configuration files such as `Dockerfile`,
`build.gradle/pom.xml`, `package.json`, `requirements.txt` or `go.mod`. `init` skips files that are larger
than 500MB.
```

> If you have multiple of those files, Skaffold will prompt you to pair your build config files
> with any images detected in your deploy configuration.

Reviewer: ```suggestion
If there are multiple build configuration files, Skaffold will prompt you to pair your build configuration files
with any images detected in your deploy configuration.
```

    - leeroy-app/kubernetes/deployment.yaml
    - leeroy-web/kubernetes/deployment.yaml
```

### Exit Codes

> When `skaffold init` fails to create a valid config, it exits with an code that depends on the error:

Reviewer: An existing `skaffold.yaml` isn't a creation problem.
```suggestion
When `skaffold init` fails, it exits with an code that depends on the error:
```

| Exit Code | Error |
| ---- | --- |
| 101 | No build configuration could be found |
| 102 | No k8s manifest could be found or generated |

> | 102 | A pre existing skaffold.yaml prevents the creation of the config |

Reviewer: Does it really prevent?  
```suggestion
| 102 | An existing skaffold.yaml was found |
```

> | 104 | Couldn't match builder with image names automatically |

Reviewer: Does this apply to `--analyze`?  Or is it only for `--analyze`?

Author: It also applies to `--analyze` although `skaffold init --analyze` should fail less often. For example, it won't fail if there's no k8s manifest, I think
